File: PS_js/baekjoon/2667.js
// let input = ["7",
//     "0110100",
//     "0110101",
//     "1110101",
//     "0000111",
//     "0100000",
//     "0111110",
//     "0111000"]
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

let N = Number(input[0]);

let maps = Array.from(Array(N), () => Array(N));

for (let i = 1; i < input.length; i++) {
    let line = input[i].split('');
    for (let j = 0; j < line.length; j++) {
        maps[i - 1][j] = line[j];
    }
}

// console.log(...maps);

let count = 0;
let array = [];
for (let i = 0; i < maps.length; i++) {
    for (let j = 0; j < maps[0].length; j++) {
        if (maps[i][j] == "1") {
            var sum = 0;
            dfs(i, j);
            array.push(sum);
            count++;
        }
    }
}

console.log(count);
let array1 = array.sort((a, b) => { return a - b; });
for (let i of array1) {
    console.log(i);
}

function dfs(i, j) {
    if (i < 0 || i >= N || j < 0 || j >= N || maps[i][j] != "1") {
        return;
    }
    sum++;
    maps[i][j] = "0";

    dfs(i, j + 1);
    dfs(i, j - 1);
    dfs(i + 1, j);
    dfs(i - 1, j);
}

File: PS.swift/2667.swift
import Foundation

let n = Int(readLine()!)!

var graph = [[Int]](repeating: [Int](repeating: 0, count: n), count: n)

for i in 0 ..< n {
    let line = readLine()!.map{Int(String($0))!}
    for j in 0 ..< n {
        graph[i][j] = line[j]
    }
}




var count = 0
func dfs(_ x: Int, _ y: Int){
    if x < 0 || x >= n || y < 0 || y >= n || graph[x][y] == 0 {
        return 
    }

    graph[x][y] = 0
    count += 1

    dfs(x, y + 1)
    dfs(x, y - 1)
    dfs(x - 1, y)
    dfs(x + 1, y)
}
var cnt = 0
var add: [Int] = []
for i in 0 ..< n {
    for j in 0 ..< n {
        if graph[i][j] == 1 {
            dfs(i, j)
            add.append(count)
            count = 0
            cnt += 1
        }
    }
}
add.sort()
print(cnt)
for i in add {
    print(i)
}